packages/ember-views/lib/system/utils.js

```diff
@@ -18,15 +22,53 @@ export const STYLE_WARNING = '' +
   'including how to disable this warning, see ' +
   'http://emberjs.com/deprecations/v1.x/#toc_binding-style-attributes.';
 
+class ConcreteBounds {
+  constructor(parent, first, last) {
+    this.parent = parent;
+    this.first  = first;
+    this.last   = last;
+  }
+
+  parentElement() {
+    return this.parent;
+  }
+
+  firstNode() {
+    return this.first;
+  }
+
+  lastNode() {
+    return this.last;
+  }
+}
+
+/**
+  @private
+  @method getViewBounds
+  @param {Ember.View} view
+*/
+export function getViewBounds(view) {
+  if (view[BOUNDS]) {
+    return view[BOUNDS];
+  } else if (view._renderNode) {
```

Comment:
if you delegated to the _renderer, this logic wouldn't need to branch, htmlbars could do one thing/glimmer the other.